library/classes/thumbnail/ThumbnailGenerator.php

```diff
@@ -0,0 +1,207 @@
+    <?php
+/**
+ * Adding thumbnails to all files
+ * User: amiel
+ * Date: 30/05/16
+ * Time: 14:55
+ */
```

Comment:
Needs a standard gpl header like you placed in above library/classes/thumbnail/Thumbnail.class.php file